src/main/java/de/yuna/berlin/nativeapp/services/http/model/HttpRequest.java

```diff
@@ -0,0 +1,322 @@
+                        }
+                    }
+                }
+            } catch (Exception ignored) {
+            }
+        }
+        return queryParams;
+    }
+
+    public boolean containsQueryParam(final String key) {
+        return queryParams.containsKey(key);
+    }
+
+    public String getQueryParam(final String key) {
+        Object value = queryParams.get(key);
+        return value != null ? value.toString() : null;
+    }
+
+    public boolean exactMatch(final String path) {
+        return this.path.startsWith(path);
+    }
+
+    public boolean match(final String pathToMatch) {
+
+        String[] partsToMatch = pathToMatch.split("/");
+        String[] parts = path.split("/");
+
+        if (exactMatch(pathToMatch) && pathToMatch.contains("{")) return true;
+
+        if (partsToMatch.length != parts.length) return false;
+
+        if (pathParams == null)
+            pathParams = new TypeMap();
+        else
+            pathParams.clear();
+        for (int i = 0; i < partsToMatch.length; i++) {
+            if (!partsToMatch[i].equals(parts[i])) {
+                if (partsToMatch[i].startsWith("{")) {
+                    String key = partsToMatch[i].substring(1, partsToMatch[i].length() - 1);
+                    pathParams.put(key, parts[i]);
+                } else {
+                    return false;
+                }
+            }
+        }
+
+        return true;
+    }
+
+
+    public TypeMap getPathParams() {
+        return pathParams;
+    }
+
+    public String getPathParam(final String key) {
+        return pathParams.get(key).toString();
+    }
+
+    public String getHeader(final String name) {
+        String value = headers.get(name).toString();
+        if (value.startsWith("[")) {
+            value = value.substring(1, value.length() - 1);
+        }
+        return value;
+    }
+
+    public TypeMap getHeaders() {
+        return headers;
+    }
+
+    public boolean containsHeader(final String name) {
+        return headers.containsKey(name.toLowerCase());
+    }
+
+    public String host() {
+        return exchange.getLocalAddress().getHostName();
+    }
+
+    public int port() {
+        return exchange.getLocalAddress().getPort();
+    }
+
+    public String protocol() {
+        return exchange.getProtocol();
+    }
+
+    public String userAgent() {
+        return getHeader(HttpHeaders.USER_AGENT);
+    }
+
+    public String getAuthorisationToken() {
+        String value = getHeader(HttpHeaders.AUTHORIZATION);
+        if (value.startsWith("Bearer ")) {
+            value = value.substring("Bearer ".length());
+        }
+        return value;
+    }
+
+    public List<String> getPreferredLanguages() {
+        String acceptLanguageHeader = getHeader(HttpHeaders.ACCEPT_LANGUAGE);
+        if (acceptLanguageHeader == null || acceptLanguageHeader.isEmpty()) {
+            return Collections.emptyList();
+        }
+
+        if (acceptLanguageHeader.startsWith("[")) {
```

Comment:
Is this valid? I couldn't find any array indication https://www.rfc-editor.org/rfc/rfc9110#field.accept-language, https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept-Language

Reply:
for test I added array in postman headers, you are right its not required.